### PR TITLE
[Editorial] The Troublesome frog

### PR DESCRIPTION
**Pull request by brebenelmihnea**
_Place an "x" in the corresponding checkbox if it is done or does not apply to this pull request._

- [x] I have tested my code.
- [x] I have added my solution according to the steps [here](https://usaco.guide/general/adding-solution#steps).
- [x] I have followed the code conventions mentioned [here](https://usaco.guide/general/adding-solution/#code-conventions).
  - I understand that if it is clear that I have not attempted to follow these conventions, my PR will be closed.
  - If changes are requested, I will re-request a review after addressing them.
- [ ] I have linked this PR to any issues that it closes.

### CODE REVIEW

**Comment by brebenelmihnea** [comment]
https://ioi.contest.codeforces.com/group/32KGsXgiKA/contest/103699/submission/268466327
image

File: solutions/platinum/ioi-02-frog.mdx [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+	}
+};
+// EndCodeSnip
+
+int main() {
+	int r, c, n;
+	cin >> r >> c >> n;
+	vector<Point> points(n);
+	array<bool, 100000000> seen;
+	auto hash = [&](const Point &p) -> int { return p.x * 10000 + p.y; };
```
Comment (SansPapyrus683) [inline solutions/platinum/ioi-02-frog.mdx:52]: you can't just gloss past this

**Review by envyaims — APPROVED** [review]
https://github.com/cpinitiative/usaco-guide/assets/79723780/380c6345-1eec-4916-b43b-a320890f59fd

File: solutions/platinum/ioi-02-frog.mdx (hunk 1 above)
Comment (brebenelmihnea) [reply solutions/platinum/ioi-02-frog.mdx:57]: added explanation

File: solutions/platinum/ioi-02-frog.mdx [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+
+int main() {
+	int r, c, n;
+	cin >> r >> c >> n;
+	vector<Point> points(n);
+	/*
+	 *	Instead of using a map - adding this way logN to the complexity -
+	 *  use a simple hashtable. The coordinates' small values allows us
+	 *  to simply concatenate the abscissa and ordinate.
+	 */
```
Comment (SansPapyrus683) [inline solutions/platinum/ioi-02-frog.mdx:55]: is unordered map too slow lol

Comment (brebenelmihnea) [reply solutions/platinum/ioi-02-frog.mdx:55]: can't use unordered map on a struct

**Pull request merged**